File: model/Exam.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ETT.model
{
    public class Exam
    {
        private Course course { get; set; }
        private List<Room> rooms { get; set; }
        private Period period { get; set; }
        private string curriculum { get; set; }
        public Exam() { }
        public Exam(Course course,List<Room> rooms ,Period period, string curriculum)
        {
            this.course = course;
            this.rooms = rooms;
            this.period = period;
            this.curriculum = curriculum;
        }

        public List<Room> getRooms() {
		    return rooms;
	    }
	    public void setRooms(List<Room> rooms) {
		    this.rooms = rooms;
	    }
        public Period getPeriod() {
            return period;
        }


        public Course getCourse() {
		    return course;
        }
        public void setCourse(Course course) {
            this.course = course;
        }

        public String getCurriculum() {
		    return curriculum;
        }

        public void setCurriculum(String curriculum) {
            this.curriculum = curriculum;
        }

        public void setPeriod(Period period) {
            this.period = period;
        }
    
    public class Builder {
            private int cost;

            private Course course;
            private List<Room> rooms;
            private Period period;
            private string curriculum;
            
            public Builder courses(Course course) {
                this.course = course;
                return this;
            }
            
            public Builder roomss(List<Room> rooms) {
                this.rooms = rooms;
                return this;
            }
            
            public Builder periods(Period period) {
                this.period = period;
                return this;

            }

            public Builder currilculums(string curriculum) {
                this.curriculum = curriculum;
                return this;
            }
            public Exam build() {
                return new Exam();
            }
        }
    }
}

File: ExaminationTimetabling/src/main/java/com/timetabling/app/model/Exam.java
package com.timetabling.app.model;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.List;

public class Exam {
	@JsonProperty("Course")
	private Course Course;
	@JsonProperty("Rooms")
	private List<Room> Rooms;
	@JsonProperty("Period")
	public Period Period;
	@JsonProperty("Curriculum")
	private String curriculum;
	
	public Exam() {}
	
	@JsonIgnore
	public Course getCourse() {
		return Course;
	}

	@JsonIgnore
	public void setCourse(Course Course) {
		this.Course = Course;
	}

	@JsonIgnore
	public List<Room> getRooms() {
		return Rooms;
	}

	@JsonIgnore
	public void setRooms(List<Room> Rooms) {
		this.Rooms = Rooms;
	}

	@JsonIgnore
	public Period getPeriod() {
		return Period;
	}

	@JsonIgnore
	public void setPeriod(Period Period) {
		this.Period = Period;
	}
	
	@JsonIgnore
	public String getCurriculum() {
		return curriculum;
	}

	@JsonIgnore
	public void setCurriculum(String curriculum) {
		this.curriculum = curriculum;
	}

	public Exam(Builder builder) {
		this.Course = builder.Course;
		this.Rooms = builder.Rooms;
		this.Period = builder.Period;
		this.curriculum = builder.curriculum;
	}
	
	public static class Builder {
		private Course Course;
		private List<Room> Rooms;
		private Period Period;
		private String curriculum;
		
		public Builder course(Course Course) {
			this.Course = Course;
			return this;
		}
		
		public Builder rooms(List<Room> Rooms) {
			this.Rooms = Rooms;
			return this;
		}
		
		public Builder period(Period Period) {
			this.Period = Period;
			return this;
		}
		
		public Builder curriculum(String curriculum) {
			this.curriculum = curriculum;
			return this;
		}
		public Exam build() {
			return new Exam(this);
		}
	}
	
}
